presto-main/src/main/java/io/prestosql/execution/CallTask.java

```diff
@@ -115,8 +122,13 @@ else if (i < procedure.getArguments().size()) {
             }
         }
 
-        // verify argument count
-        if (names.size() < positions.size()) {
+        // verify provided argument count
+        int providedArgumentsCount = toIntExact(procedure.getArguments().stream()
+                .filter(Argument::isRequired)
+                .filter(argument -> names.containsKey(argument.getName()))
+                .count());
+
+        if (providedArgumentsCount < requiredArgumentsCount) {
```

Comment:
`providedArgumentsCount` is not a good name, since we count only those provided args which are required

also, we miss opportunity to say which required argument is missing.

maybe:
```
procedure.getArguments().stream()
        .filter(Argument::isRequired)
        .filter(argument -> !names.containsKey(argument.getName()))
        .findFirst()
        .ifPresent(argument -> {
            throw semanticException(INVALID_ARGUMENTS, call, "No value provided for a required argument: " + argument.getName());
        });
```

Reply:
Good idea :) That suggests solution to the end-user